### ISSUE TITLE
Migration guide to 0.30?

### ISSUE THREAD

**Issue by AndreasBackx** [Smithay/wayland-rs#625]
It's great to see the 0.30 making some great and deserved changes. I'm using it for a new project and want to migrate an older project: https://github.com/AndreasBackx/waycorner. A few years ago I used the Smithay book, but it seems that it's out-of-date, though there is an "unpublished" 0.30 branch, it is 1 year old when 0.30 seems newer than that so I'm unsure it's up-to-date.

Regardless, where does one start to read up on how to migrate from the older spec? I enjoyed the book a few years ago, though I haven't entirely kept up-to-date with what's happening so I could also use with a refresher. 😅

Thanks again for the great tools!

**Comment by elinorbgr** [comment]
Yeah, the smithay book has been quite abandoned for some time now. We just don't manage to secure time to work on it, which is a shame.

I'm sorry for the lack of tutorial, but currently the best you'll find for 0.30 are the API docs. I tried to keep module-level and crate-level documentation detailed enough to present the main principles on how the library should be used.

The main difference in terms of API is how the event dispatching is now handled: your app is mostly expected to have a single main `State` type, on which several instances of the `Dispatch<_>` trait need to be implemented for handling events.
